akka-actor/src/main/scala/akka/actor/dungeon/Dispatch.scala

```diff
@@ -123,27 +123,48 @@ private[akka] trait Dispatch { this: ActorCell ⇒
 
   def sendMessage(msg: Envelope): Unit =
     try {
-      if (system.settings.SerializeAllMessages) {
-        val unwrapped = (msg.message match {
-          case DeadLetter(wrapped, _, _) ⇒ wrapped
-          case other                     ⇒ other
-        }).asInstanceOf[AnyRef]
-        if (!unwrapped.isInstanceOf[NoSerializationVerificationNeeded]) {
-          val s = SerializationExtension(system)
-          val serializer = s.findSerializerFor(unwrapped)
-          val bytes = serializer.toBinary(unwrapped)
-          serializer match {
-            case ser2: SerializerWithStringManifest ⇒
-              val manifest = ser2.manifest(unwrapped)
-              s.deserialize(bytes, serializer.identifier, manifest).get != null
-            case _ ⇒
-              s.deserialize(bytes, unwrapped.getClass).get
-          }
+      val msgToDispatch =
+        if (!system.settings.SerializeAllMessages) {
```

Comment:
I didn't measure, just went with the old "the most important/common thing first" rule, the branches are so small and if it is disabled then the other branch will never be inlined as it is never called. I'll invert.